File: 2015/python/day1.py
with open('../input/day1.txt') as file:
    input = file.readline()

floor = 0
position = 1
enteredBasementPosition = 0
for instruction in input:
    if instruction == '(':
        floor += 1
    elif instruction == ')':
        floor -= 1
    if floor == -1 and enteredBasementPosition == 0:
        enteredBasementPosition = position
    position += 1

print('partOne = ' + str(floor))
print('partTwo = ' + str(enteredBasementPosition))

# $ python3 day1.py
# partOne = 74
# partTwo = 1795

File: 2015/rust/day1.rs
use std::fs;

fn main() {
    let mut floor = 0;
    let mut basement_index = 0;

    let contents = fs::read_to_string("../input/day1.txt").unwrap();
    
    for (i, c) in contents.chars().enumerate() {
        if c == '(' {
            floor += 1;
        } else {
            floor -= 1;
        }

        if floor == -1 && basement_index == 0 {
            basement_index = i + 1;
        }
    }

    println!("partOne = {}\npartTwo = {}", floor, basement_index);
}